README.md

```diff
@@ -16,3 +52,161 @@ pip install cookiecutter
 ```sh
 cookiecutter gh:elixir-cloud-aai/cookiecutter-python
 ```
+
+> **Note**: If you're using just `cookiecutter`, manually remove
+  `update-template` job from `update.yaml` github action workflow file as
+  `cookiecutter` doesn't support autosync like cruft.
+
+## Configurations
+
+### Package manager - Poetry
+
+This project uses [Poetry][poetry-org] as a package manager. Check out the
+commands at the [official documentation][poetry-docs].
+
+### Linters and formatters - Ruff
+
+To lint and format `Python` code files, it uses [Ruff][ruff], the default
+configuration is set in the `pyproject.toml` file.
+
+```toml
+select = [
+  "B", # flake8-bugbear
+  "E", # pycodestyle
+  "F", # Pyflakes
+  "I", # isort
+  "PL", # pylint
+  "SIM", # flake8-simplify
+  "UP", # pyupgrade
+]
+```
+
+To configure it to your needs, refer to the [rules documentation][rudff-linter]
+, and for formatter configuration, refer to the
+[configuration documentation][ruff-formatter].
+
+### Spell checker - Typos
+
+If you want to ignore certain words, add them to the `pyproject.toml` file,
+under the `tool.typos.default.extend-words` key.
+
+```toml
+[tool.typos.default.extend-words]
+mke = 'mke'
+```
+
+For further configuration, refer to the [typos docs][typos-docs].
+
+### Static type checker - Mypy
+
+Change the configuration in `pyproject.toml` file, for further information refer to
+the [documentation][mypy-config].
+
+### Documentation - Sphinx
+
+Configuration for Sphinx is in the `docs/source/conf.py` file, for further info refer
+to the [Sphinx documentation][sphinx-docs]. The default configuration uses
+[furo theme][furo] and [ReadTheDocs][rtd] to host the documentation, you can
+change the configuration using `/docs/source/conf.py` and `.readthedocs.yml`
+file.
+
+> **Note**: Make sure to setup the `ReadTheDocs` account and add the project to
+  the account to host the documentation.
+
+### Testing - pytest and pytest-cov
+
+Generate a coverage report using `pytest-cov` and uploads it to codecov.io.
+
+> **Note**: Make sure to setup the `Codecov` account and add the project to the
+  account to host the coverage report and add github [token](#cicd---github-actions).
+
+## CI/CD - GitHub Actions
+
+Here are the GitHub Actions Secrets that need to be included in the repository
+settings:
+
+- `PYPI_PASSWORD`: PyPI account password.
+- `CODECOV_TOKEN`: Codecov token.
+- `AUTO_UPDATE_GITHUB_TOKEN`: GitHub token with permissions to write to the repository.
```

Comment:
Can you please specify _precisely_ what fine-grained permissions are needed and how to set this up (you can refer to [this doc](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) for instructions on the general procedure of how to add personal access tokens)?

Reply:
Hey can you please look at [this](https://cruft.github.io/cruft/#:~:text=always%20submitted%20upstream.-,Automating%20updates%20with%20GitHub%20Actions,-If%20you%20have) and specially the comment, will this be enough, the thing is idk how to test it at organiszation level :sweat:.

Let me know if this works then I'll add the same to the readme.

Reply:
I think it will be fine. Let's just try it an see how it goes. If necessary, we can always change it later.

Reply:
added guide in the readme of the repo anyways.